chautari-rental-service.yml

```diff
@@ -0,0 +1,938 @@
+      properties:
+        title:
+          type: string
+          enum: [Ms, Mr, Miss]
+        firstName:
+          type: string
+        middleName:
+          type: string
+        lastName:
+          type: string
+        status:
+          type: string
+          enum: [single, married]
+    SetupResponse:
+      type: object
+      properties:
+        user-id:
+          type: string
+          format: UUID
+        message:
+          type: string
+    RentalItemRequest:
+      type: object
+      required:
+        - saf
+      properties:
+        owner-id:
+          type: string
+        title:
+          type: string
+        rentOf:
+          type: string
+          enum: [room]
+        availability:
+          $ref: '#/components/schemas/Availability'
+        location:
+          $ref: '#/components/schemas/Address'
+        price:
+          $ref: '#/components/schemas/RentalItemPrice'
+        contact:
+          $ref: '#/components/schemas/Contact'
+        preferences:
+          type: array
+          items:
+            $ref:  '#/components/schemas/RentalItemPreference'
+        features:
+          type: object
+          properties:
+            available:
+              type: array
+              items:
+                type: string
+            not-available:
+              type: array
+              items:
+                type: string
+        postedOn:
+          type: string
+        image-ids:
+          type: array
+          items:
+            type: string
+            example: 123423241
+    RentalItemPreference:
+      type: object
+      required:
+        - preference
+      properties:
+        preference:
+          type: string
+        priority:
+          type: string
+          enum: [high, medium, low]
+    RentalItemPrice:
+      type: object
+      properties:
+        value:
+          type: number
+        type:
+          type: string
+          enum: [negotiable, non-negotiable]
+    Contact:
+      type: object
+      required:
+        - phoneNumber
+      properties:
+        phoneNumber:
+          type: string
+        email:
+          type: string
+    Availability:
+      type: object
+      properties:
+        available:
+          $ref: '#/components/schemas/Duration'
+        not-available:
+          type: array
+          items:
+            $ref: '#/components/schemas/Duration'
+    Duration:
+      type: object
+      properties:
+        start-date:
+          type: string
+        end-date:
+          type: string
+    RentalItemSummary:
+      type: object
+      required:
+        - title
+        - price
+        - rentOf
+        - location
+        - image
+      properties:
+        title:
+          type: string
+        price:
+          $ref: '#/components/schemas/RentalItemPrice'
+        rentOf:
+          type: string
+        location:
+          $ref: '#/components/schemas/Address'
+        image-id:
+          type: string
+    RentalItemDetail:
```

Comment:
we need to track the rental item status like new, inProgress, rented or something like this ?